### PR TITLE
feat: Adds an endpoint and handlers to change user account password

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

- Adds an endpoint and handlers to change user account password
- Adds test cases

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: internal/api/handlers.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ func NewGoCertRouter(env *Environment) http.Handler {
 	apiV1Router.HandleFunc("GET /accounts/{id}", GetUserAccount(env))
 	apiV1Router.HandleFunc("GET /accounts", GetUserAccounts(env))
 	apiV1Router.HandleFunc("POST /accounts", PostUserAccount(env))
+	apiV1Router.HandleFunc("POST /accounts/{id}", ChangeUserAccountPassword(env))
```
Comment (kayra1) [inline internal/api/handlers.go:37]:
```suggestion
	apiV1Router.HandleFunc("POST /accounts/{id}/change_password", ChangeUserAccountPassword(env))
```

**Review by kayra1 — CHANGES_REQUESTED** [review]

File: internal/api/handlers.go [hunk 2]
```diff
@@ -360,6 +361,34 @@ func PostUserAccount(env *Environment) http.HandlerFunc {
...
+		var user certdb.User
+		if err := json.NewDecoder(r.Body).Decode(&user); err != nil {
+			logErrorAndWriteResponse("Invalid JSON format", http.StatusBadRequest, w)
+			return
+		}
+		if user.Password == "" {
+			logErrorAndWriteResponse("Password is required", http.StatusBadRequest, w)
+			return
+		}
+		ret, err := env.DB.UpdateUser(id, user.Password)
```
Comment (kayra1) [inline internal/api/handlers.go:376]: should we have validations like more than 8 character or at least one capital letter and symbol here?

Comment (saltiyazan) [reply internal/api/handlers.go:394]: Sure, and done it at user creation too.

Comment (DanielArndt) [reply internal/api/handlers.go:394]: 8? We still in the 90's or something? :laughing:

Comment (saltiyazan) [reply internal/api/handlers.go:394]: Isn't that good enough for a minimum requirement?
I increased the length of the generated password. @DanielArndt

**Comment by saltiyazan** [comment]
My latest commit is taking too long to appear for some reason.

Screenshot 2024-07-05 at 14 54 20

**Comment by kayra1** [comment]
first time in my life i'm seeing this

File: internal/api/handlers.go (hunk 2 above)
Comment (kayra1) [reply internal/api/handlers.go:421]: I think its fine fyi :D

**Review by kayra1 — APPROVED** [review]
nice

**Pull request merged**